Packs/UnifiVideoNVR/Integrations/UnifiVideo/UnifiVideo.py

```diff
@@ -0,0 +1,194 @@
+            rval, frame = vc.read()
+            c = c + 1
+            if c == int(demisto.args()['frame']):
+                cv2.imwrite('/tmp/snapshot.jpg', frame)
+                break
+        vc.release()
+        f = open("/tmp/snapshot.jpg", "r")
+        output = f.read()
+        filename = "snapshot.jpg"
+        file = fileResult(filename=filename, data=output)
+        if r_type == 'jpg':
+            file['Type'] = entryTypes['image']
+        demisto.results(file)
+    sys.exit(0)
+
+if demisto.command() == 'get_recording_list':
+    uva = UnifiVideoAPI(api_key=demisto.params()["api_key"], addr=demisto.params()["addr"],
+                        port=demisto.params()["port"], schema=demisto.params()["schema"],
+                        verify_cert=demisto.params()["verify_cert"])
+    for rec in uva.recordings:
+        print(rec)
+    sys.exit(0)
+
+if demisto.command() == 'create_inc':
+    lastRun = demisto.getLastRun()
+    print(lastRun)
+    sys.exit(0)
+
+if demisto.command() == 'get_snapshot_at_frame':
+    vc = cv2.VideoCapture('recording.mp4')
+    c = 1
+
+    if vc.isOpened():
+        rval, frame = vc.read()
+    else:
+        rval = False
+
+    while rval:
+        rval, frame = vc.read()
+        c = c + 1
+        if c == 500:
+            cv2.imwrite(str(c) + '.jpg', frame)
+            break
+    vc.release()
+    sys.exit(0)
+
+if demisto.command() == 'fetch-incidents':
+    start_time_of_int = str(datetime.now())
+    uva = UnifiVideoAPI(api_key=demisto.params()["api_key"], addr=demisto.params()["addr"], port=demisto.params()[
+                        "port"], schema=demisto.params()["schema"], verify_cert=demisto.params()["verify_cert"])
+
+    # And retrieve it for use later:
+    last_run = demisto.getLastRun()
+    # lastRun is a dictionary, with value "now" for key "time".
+    # JSON of the incident type created by this integration
+    inc = []
+    day_ago = datetime.now() - timedelta(days=1)
+    start_time = day_ago
+    if last_run and last_run.has_key('start_time'):
+        start_time = last_run.get('start_time')
+    for rec in uva.recordings:
+        incident = {}
+        datetime_object = datetime.strptime(str(rec.start_time), '%Y-%m-%d %H:%M:%S')
+        for camera in uva.cameras:
+            cam_id = uva.get_camera(camera.name)
+            if cam_id._id in rec.cameras:
+                camera_name = camera.name
+
+        incident = {
+            'name': rec.rec_type,
+            'occurred': datetime_object.isoformat() + "Z",
+            'rawJSON': json.dumps({"event": rec.rec_type, "ubnt_id": rec._id, "camera_name": camera_name, "integration_lastrun": str(start_time), "start_time": str(rec.start_time), "stop_time": str(rec.end_time)})
+        }
+        try:
+            if not isinstance(start_time, datetime):
+                start_time = datetime.strptime(start_time, '%Y-%m-%d %H:%M:%S.%f')
+            if datetime_object > start_time:
+                inc.append(incident)
```

Comment:
Suggestion - It will be more efficient to check if the incident is valid before creating it.

Reply:
done